src/native/fs/fs_napi.cpp

```diff
@@ -213,6 +214,112 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * LStat is an fs op
+ */
+struct LStat : public FSWorker
+{
+    std::string _path;
+    struct stat _stat_res;
+
+    LStat(const Napi::CallbackInfo& info)
+        : FSWorker(info)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "LStat " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        int r = lstat(_path.c_str(), &_stat_res);
+        if (r) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG1("FS::LStat::OnOK: " << DVAL(_path) << DVAL(_stat_res.st_ino) << DVAL(_stat_res.st_size));
+        Napi::Env env = Env();
+        auto res = Napi::Object::New(env);
+        res["dev"] = Napi::Number::New(env, _stat_res.st_dev);
```

Comment:
How about creating a common function to do all this work to read from stat_res and populate res?